pkg/dmsclient/instrumented_client.go

```diff
@@ -0,0 +1,125 @@
+package dmsclient
```

Comment:
This file seems like a really heavy way to instrument the dms client. Why not simply wrap the guts of the `do` func? I can see it'll be a little bit tricky to get the `method` label in place for `FindSnitchesByName`, if we even care to distinguish that from `ListAll`, but it should be doable.

Reply:
IMO it's the same amount of complexity. In the `do()` method the path and payload will have to be inspected to determine which method is being called. I didn't want to change a lot of the existing code and instead chose to wrap it.

Reply:
It's not about complexity -- I agree it would be doing the same thing either way, as it should. My concern is the code repetition, which carries with it the usual maintainability issues. There's also no need for the uninstrumented client to exist anymore.